zipline/assets/futures.py

```diff
@@ -0,0 +1,128 @@
+from pandas import Timedelta
+
+from zipline.finance.trading import TradingEnvironment
+
+
+class FutureChain(object):
+    """ Allows users to look up future contracts.
+
+    Parameters
+    ----------
+    trading_algorithm : TradingAlgorithm
+    root_symbol : str, optional
+    as_of_date : pandas.Timestamp
+    """
+
+    def __init__(self, trading_algorithm, root_symbol='', as_of_date=None):
+        self.root_symbol = root_symbol.upper()
```

Comment:
This conversion should probably be done by the context that constructs the FutureChain, not by the chain itself.